File: vt21-sysa14-pk-pa1-vt21-sysa14-it-s-not-just-for-boys/JavaClientAssignment6/src/Client6/Errorhandler.java
package Client6;

import java.rmi.RemoteException;
import java.sql.SQLException;

import javax.xml.soap.SOAPException;

public class Errorhandler {

	public String handleException(Exception ex) {
		String errorMessage;
			
		if (ex instanceof NullPointerException) {
			errorMessage = "This object doesn't exist";
		}

		else if (ex instanceof IndexOutOfBoundsException) {
			errorMessage = "Coudn't find what you are looking for";
		}
		else if (ex instanceof RemoteException) {
			errorMessage = "Connection with web service failed";
		}

		else if (ex instanceof SQLException) {
			int errorCode = ((SQLException) ex).getErrorCode();
			switch (errorCode) {

			case 17:
				errorMessage = "Problem with the connection to the database";
				break;

			case 0:
				errorMessage = "Connection failed, timeout reached";
				break;
				
			default:
				errorMessage = "An unexpected error has occured";
				break;
			}
		} else {
			errorMessage = "An unexpected error has occured";
		}
		
		return errorMessage; 
	}
}

File: WSClient_Assignment3/ErrorHandler.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WSClient_Assignment3
{
    public class ErrorHandler
    {

        public string HandleException(Exception ex)
        {
            string errorMessage= "";


            if (ex.GetType() == typeof(System.IO.FileNotFoundException))
            {
                errorMessage = "This file doesn't exist";
            }

            else if (ex.GetType() == typeof(System.Net.WebException))
            {
                errorMessage = "Connection to web server failed";
            }

            if (ex.GetType() == typeof(NullReferenceException))
            {

                errorMessage = "This file doesn't exist";
            }

            else if (ex.GetType() == typeof(IndexOutOfRangeException))
            {
                errorMessage = "Coudn't find what you are looking for";
            }

            else
            {
                errorMessage = "An unexpected error has occured";
            }


            return errorMessage;

        }

    }
}
